README.md

```diff
@@ -25,24 +25,76 @@ GovTech, and LegalTech. The Exonum framework is oriented towards creating
-  * [Time service](services/time/README.md).
-* Examples
-  * [Cryptocurrency](examples/cryptocurrency/README.md).
-  * [Cryptocurrency-advanced](examples/cryptocurrency-advanced/README.md).
-  * [Timestamping](examples/timestamping/README.md).
+## Contents
 
-See individual projects readme for the details.
+This is the main Exonum repository containing the bulk of Rust crates
+used in Exonum. Rust crates for Exonum are intended to be reasonably
+small and reusable, hence there is relatively large number of them.
 
-If you are using Exonum in your project and want to be listed on our website &
-GitHub list — write us a line to <contact@exonum.com>.
+### Main Crates
+
+- [Core library](exonum/README.md)
+- [Node implementation](exonum-node/README.md)
+- [Node CLI](cli/README.md)
+
+### Upstream Dependencies
+
+- [Cryptographic library](components/crypto/README.md)
+- [Database backend for merkelized data structures](components/merkledb/README.md)
+- [Key management](components/keys/README.md)
+- [Derive macros](components/derive/README.md)
+- [Protobuf helpers](components/proto/README.md)
+- [Protobuf support for build scripts](components/build/README.md)
+- [High-level HTTP API abstraction](components/api/README.md)
+
+### Tools for Building Services
+
+- [Rust runtime](runtimes/rust/README.md)
+- [Testing framework](test-suite/testkit/README.md)
+
+### Services and Node Plugins
+
+- [Explorer service](services/explorer/README.md) and [explorer library](components/explorer/README.md)
+- [Middleware service](services/middleware/README.md)
+- [Supervisor service](services/supervisor/README.md)
+- [Time oracle service](services/time/README.md)
+- [System API plugin](components/system-api/README.md)
+
+### Examples
+
+- [Cryptocurrency](examples/cryptocurrency/README.md)
+- [Advanced cryptocurrency](examples/cryptocurrency-advanced/README.md)
+- [Timestamping](examples/timestamping/README.md)
+- [Sample runtime implementation](examples/sample_runtime/README.md)
+
+## Versioning Policy
+
+Exonum crates follow [semantic versioning](https://semver.org/).
+
+The `exonum` crate and its re-exported dependencies
+(`exonum-crypto`, `exonum-merkledb` and `exonum-keys`) are released
+at the same time; their version is considered *the* version of the Exonum framework.
+On the other hand, the crates downstream to `exonum` (e.g., `exonum-node`)
+or independent of it (e.g., `exonum-api`) may evolve at different speeds,
+including major releases not tied to a major Exonum release.
+
+Throughout the Exonum codebase, certain APIs are described in the API docs
+as unstable or experimental. Such APIs may be removed or changed
+in a semantically non-breaking release (for example, a minor release)
+of the corresponding crate.
+
+## Contributing
+
+To contribute to Exonum, please see [CONTRIBUTING](CONTRIBUTING.md).
+
+## See Also
 
-## Other languages support
+Some Exonum stuff that is *not* in this repository:
```

Comment:
Should we list `exonum-launcher` here as well?

Reply:
I'm open to suggestions 😄